plugins/Toolbox/src/Toolbox.py

```diff
@@ -305,9 +314,87 @@ def install(self, file_path: str) -> None:
+        if package_used_materials or package_used_qualities:
+            # Set up "uninstall variables" for resetMaterialsQualitiesAndUninstall
+            self._package_id_to_uninstall = package_id
+            package_info = self._package_manager.getInstalledPackageInfo(package_id)
+            self._package_name_to_uninstall = package_info.get("display_name", package_info.get("package_id"))
+            self._package_used_materials = package_used_materials
+            self._package_used_qualities = package_used_qualities
+            # Ask change to default material / profile
+            if self._confirm_reset_dialog is None:
+                self._confirm_reset_dialog = self._createDialog("ToolboxConfirmUninstallResetDialog.qml")
+            self.uninstallVariablesChanged.emit()
+            self._confirm_reset_dialog.show()
+        else:
+            # Plain uninstall
+            self.uninstall(package_id)
+
+    @pyqtProperty(str, notify = uninstallVariablesChanged)
+    def pluginToUninstall(self):
+        return self._package_name_to_uninstall
+
+    @pyqtProperty(str, notify = uninstallVariablesChanged)
+    def uninstallUsedMaterials(self):
+        return "\n".join(["%s (%s)" % (str(global_stack.getName()), material) for global_stack, extruder_nr, material in self._package_used_materials])
+
+    @pyqtProperty(str, notify = uninstallVariablesChanged)
+    def uninstallUsedQualities(self):
+        return "\n".join(["%s (%s)" % (str(global_stack.getName()), quality) for global_stack, extruder_nr, quality in self._package_used_qualities])
+
+    @pyqtSlot()
+    def closeConfirmResetDialog(self):
+        if self._confirm_reset_dialog is not None:
+            self._confirm_reset_dialog.close()
+
+    ##  Uses "uninstall variables" to reset qualities and materials, then uninstall
+    #   It's used as an action on Confirm reset on Uninstall
+    @pyqtSlot()
+    def resetMaterialsQualitiesAndUninstall(self):
+        application = CuraApplication.getInstance()
+        material_manager = application.getMaterialManager()
+        quality_manager = application.getQualityManager()
+        machine_manager = application.getMachineManager()
+
+        for global_stack, extruder_nr, container_id in self._package_used_materials:
+            default_material_node = material_manager.getDefaultMaterial(global_stack, extruder_nr, global_stack.extruders[extruder_nr].variant.getName())
+            machine_manager.setMaterial(extruder_nr, default_material_node, global_stack = global_stack)
+        for global_stack, extruder_nr, container_id in self._package_used_qualities:
+            default_quality_group = quality_manager.getDefaultQualityType(global_stack)
+            machine_manager.setQualityGroup(default_quality_group, global_stack = global_stack)
+
+        self._markPackageMaterialsAsRemove(self._package_id_to_uninstall)
```

Comment:
Could have a more clear name to make it easier to know if stuff is being marked "as removed" or "as removable" or what. "remove" is kind of ambiguous.